### ISSUE TITLE
Intermittent 502 errors on OpenRelay.xyz

### ISSUE THREAD

**Issue by AusIV** [NoteGio/openrelay#16]
Intermittently, requests made to api.openrelay.xyz come back with a 502 error.

**Comment by AusIV** [comment]
I've only ever seen this on requests that go to the search API microservice, never on requests to the ingest API microservice.

**Comment by AusIV** [comment]
Adding logging statements to the search microservice seems to have stopped the 502s. Even if the logging statements are below the log level that's actually being logged, the 502s aren't happening.

I'm going to leave this open for now, in case the issue resurfaces.

**Comment by AusIV** [comment]
Migrating from Docker Swarm to Amazon ECS seems to have resolved this issue. I'm going to hold off on closing it to see if it rears its head again, but it looks promising.

Basically, it appears that this may have been an issue with Docker swarm's overlay network. I found some references to other people having the same issue. I think it could have been resolved by using the DNSRR method of exposing the port, but there were other motivations for migrating from Swarm to ECS.

ECS just exposes the host port straight to the loadbalancer, so there's no software-level networking in the process.
